### ISSUE TITLE
All grains fail during OnActiveAsync

### ISSUE THREAD

**Issue by FranKieCZ** [OrleansContrib/OrleansDashboard#17]
Hi,

I've recently updated our App to Orleans 1.3.0 and it works fine. However, I wanted to try your Dashboard (especially the latency and throughout metrics, which I love! :) ) but when i add Dasboard to the project, any call to any Grain fail.

```
Log:
[2016-10-20 13:00:46.826 GMT    28  INFO    100000  SiloControl 135.128.216.44:11111]   GetSimpleGrainStatistics    
[2016-10-20 13:00:46.822 GMT    21  ERROR   100513  Catalog 135.128.216.44:11111]   !!!!!!!!!! Error calling grain's OnActivateAsync() method - Grain type = XXXXXX.Statistics.QueueLengthHistoryMaintenanceGrain Activation = [Activation: S135.128.216.44:11111:214664102*grn/CCD89FBB/00000000@e699b3ac #GrainType=XXXXX.Statistics.QueueLengthHistoryMaintenanceGrain Placement=RandomPlacement State=Activating]   
Exc level 0: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.

Server stack trace: 
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Orleans.InterceptedMethodInvokerCache.GetInterfaceToImplementationMap(Int32 interfaceId, Type implementationType)
   at Orleans.InterceptedMethodInvokerCache.CreateInterceptedMethodInvoker(Type implementationType, Int32 interfaceId, IGrainMethodInvoker invoker)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Orleans.InterceptedMethodInvokerCache.GetOrCreate(Type implementationType, Int32 interfaceId, IGrainMethodInvoker invoker)
   at Orleans.Runtime.InsideRuntimeClient.InvokeWithInterceptors(IAddressable target, InvokeMethodRequest request, IGrainMethodInvoker invoker)
   at Orleans.Runtime.InsideRuntimeClient.<Invoke>d__57.MoveNext()

Exception rethrown at [0]: 
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at XXXXX.Statistics.QueueLengthHistoryMaintenanceGrain.<ReminderInterval>d__8.MoveNext() in C:\XXXXXX\Statistics\QueueLengthHistoryMaintenanceGrain.cs:line 36
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at XXXXXX.ReminderRepeaterBaseGrain.<OnActivateAsync>d__6.MoveNext() in C:\XXXXXX\ReminderRepeaterBaseGrain.cs:line 47
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Orleans.Runtime.Catalog.<CallGrainActivate>d__66.MoveNext()
[2016-10-20 13:00:46.879 GMT    28  WARNING 100534  Catalog 135.128.216.44:11111]   Failed to InvokeActivate for [Activation: S135.128.216.44:11111:214664102*grn/CCD89FBB/00000000@e699b3ac #GrainType=XXXXXX.Statistics.QueueLengthHistoryMaintenanceGrain Placement=RandomPlacement State=Invalid].  
Exc level 0: System.Collections.Generic.KeyNotFoundException: The given key was not present in the dictionary.
```

In our project we use Bootstrap to init IoC container (Unity), we plan to change it to the way how DI works in Orleans now but we didn't have time to do it so (we upgraded from Orleans 1.0.10 where DI was not in a way how it's now).

If I comment out this line from config:

```
 <Provider Type="OrleansDashboard.Dashboard" Name="Dashboard" />
```

it works fine so it seems that Dashboard Bootstrap somehow mess things up?

**Comment by richorama** [comment]
Have you tried adding `Microsoft.Orleans.OrleansCodeGenerator` to the host project? It seemed to fix #6.

**Comment by FranKieCZ** [comment]
Yes, OrleansCodeGenerator is in both Host and Client parts of the project.

Just one note, i'm running it on top of .NET 4.6.1 if it could cause any issues...?
